### ISSUE TITLE
isValid() doesn't work on nested fields

### ISSUE THREAD

**Issue by joedj** [caolan/forms#153]
Validation errors are detected in nested fields, but isValid() returns true and handle() fires the success callback rather than the error callback.  Is this supposed to be a supported feature?

Test case here: https://gist.github.com/joedj/00845444f6185de4b93a

```bash
git clone https://gist.github.com/00845444f6185de4b93a.git nested-fields
cd nested-fields
npm install
node test.js
```

**Comment by ljharb** [comment]
Released as v1.1.1.

Thanks a lot for the amazingly helpful gist, test case, and issue text - if all bugs were submitted this way there wouldn't be any left :-)

**Comment by joedj** [comment]
Wow, that was quick.  Thanks!
